src/components/ToggleSwitch/index.tsx

```diff
@@ -0,0 +1,206 @@
+import * as React from 'react';
+import {
+  StyleSheet,
+  Animated,
+  TouchableOpacity,
+  Easing,
+  StyleProp,
+  Text,
+  TextStyle,
+  Platform,
+  TouchableOpacityProps
+} from 'react-native';
+import dimen from '../../utils/dimen';
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: 'row',
+    alignItems: 'center'
+  },
+  basicStyle: {
+    height: dimen.normalizeDimen(16),
+    width: dimen.normalizeDimen(16),
+    borderRadius: dimen.normalizeDimen(16 / 2),
+    backgroundColor: '#2C67BC'
+  },
+  eahcStyles: {
+    fontSize: dimen.normalizeDimen(12),
+    color: '#f4f3f4',
+    position: 'absolute',
+    top: dimen.normalizeDimen(2),
+    left: dimen.normalizeDimen(3)
+  },
+
+  eahcStylesOf: {
+    fontSize: dimen.normalizeDimen(12),
+    color: '#f4f3f4',
+    position: 'absolute',
+    top: dimen.normalizeDimen(2),
+    right: dimen.normalizeDimen(3)
+  },
+  mainStyes: {
+    borderRadius: dimen.normalizeDimen(12),
+    backgroundColor: '#F5F5F5',
+    height: dimen.normalizeDimen(20),
+    width: dimen.normalizeDimen(42),
+    padding: dimen.normalizeDimen(2),
+    ...Platform.select({
+      ios: {
+        shadowColor: 'rgba(0,0,0,0.25)',
+        shadowOffset: {width: 0, height: 0},
+        shadowOpacity: dimen.normalizeDimen(1),
+        shadowRadius: dimen.normalizeDimen(1)
+      },
+      android: {
+        elevation: dimen.normalizeDimen(1)
+      }
+    })
+  },
+  labelLeft: {
+    marginRight: 5,
+    color: '#FFFFFF',
+    fontSize: 12,
+    fontWeight: '400'
+  },
+  labelRight: {
+    marginLeft: 5,
+    color: '#FFFFFF',
+    fontSize: 12,
+    fontWeight: '400'
+  }
+});
+
+interface ToggleSwitchProps {
+  value?: boolean;
+  labelLeft?: string;
+  labelRight?: string;
+  circleActiveColor?: string;
+  circleInActiveColor?: string;
+  backgroundActive?: string;
+  backgroundInactive?: string;
+  styleLabelLeft?: StyleProp<TextStyle>;
+  styleLabelRight?: StyleProp<TextStyle>;
+  activeTextColor?: string;
+  inactiveTextColor?: string;
+  conatinerStyle?: TouchableOpacityProps['style'];
+  onValueChange?: (value: boolean) => void;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is a typo in the prop name `conatinerStyle`. It should be `containerStyle`.
```diff
-   conatinerStyle?: TouchableOpacityProps['style'];
+   containerStyle?: TouchableOpacityProps['style'];
  onValueChange?: (value: boolean) => void;
```

102:
There is a typo in the variable name `conatinerStyle`. It should be `containerStyle`.
```diff
-   conatinerStyle
+   containerStyle
```

153:
There is a typo in the variable name `conatinerStyle`. It should be `containerStyle`.
```diff
-       style={[styles.container, conatinerStyle]}
+       style={[styles.container, containerStyle]}
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->